### PR TITLE
docs: rework "admin/authentication" page + PKI authentication docs

### LINKED ISSUE #9188: Add docs for OIDC PKI Auth

### PR DESCRIPTION
**Pull request by bpmct**
@Emyrk brought up that this page was getting a bit out of hand, and I agree. Would appreciate folks thoughts on the new format. 

Closes https://github.com/coder/coder/issues/9188

## Before merging

- [ ] Use local images instead of githubusercontent (it was easier to preview with public URLs https://github.com/coder/coder.com/issues/273)
- [ ] Ensure there aren't broken links in the docs/product
- [ ] Final sweep for grammar

I'd appreciate a review in the meantime.

### CODE REVIEW

File: docs/admin/auth.md [hunk 1]
```diff
@@ -94,9 +92,9 @@ Your OIDC provider will ask you for the following parameter:
 
 - **Redirect URI**: Set to `https://coder.domain.com/api/v2/users/oidc/callback`
 
-### Step 2: Configure Coder with the OpenID Connect credentials
+### Step 2 (Client Secret Auth): Configure Coder with the OpenID Connect credentials
 
-Navigate to your Coder host and run the following command to start up the Coder
+The most common way to authenticate with OIDC providers is with a `client_secret`. Navigate to your Coder host and run the following command to start up the Coder
 server:
```
Comment (matifali) [inline docs/admin/auth.md:99]: What do you think about adding subheadings?
1. CLI
2. System Service
3. Helm

File: docs/admin/auth.md [hunk 2]
```diff
@@ -138,6 +136,74 @@ To upgrade Coder, run:
...
+You must upload the public key (the certificate) to your OIDC provider.
+Reference the documentation provided by your provider on how to do this. Depending on the provider, the name for this feature varies.
+
+- <!-- Azure --> Authentication certificate credentials
+- <!-- Okta --> JWT for Client Authentication
+- <!-- Auth0 --> Authenticate with Private Key JWT
+
+Once the key and certificate are uploaded, you can run Coder with the files. Navigate to your Coder host and run the following command to start up the Coder server:
+
+```console
```
Comment (matifali) [inline docs/admin/auth.md:171]: the same concept of subheadings here.

File: docs/admin/auth/github.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+## Configuring
+
+### Step 1: Configure the OAuth application in GitHub
+
+First, [register a GitHub OAuth app](https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/). GitHub will ask you for the following Coder parameters:
+
+- **Homepage URL**: Set to your Coder deployments [`CODER_ACCESS_URL`](https://coder.com/docs/v2/latest/cli/server#--access-url) (e.g. `https://coder.example.com`)
+- **User Authorization Callback URL**: Set to `https://coder.example.com/api/v2/users/oauth2/github/callback`
+
+> Note: If you want to allow multiple coder deployments hosted on subdomains e.g. coder1.example.com, coder2.example.com, to be able to authenticate with the same GitHub OAuth app, then you can set **User Authorization Callback URL** to the `https://example.com`
```
Comment (matifali) [inline docs/admin/auth/github.md:16]:
```suggestion
> [!NOTE] 
> If you want to allow multiple coder deployments hosted on subdomains, e.g., `coder1.example.com`, `coder2.example.com`, to be able to authenticate with the same GitHub OAuth app, then you can set **User Authorization Callback URL** to the `https://example.com`
```

File: docs/admin/auth/adfs.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Active Directory Federation Services OIDC with Coder (ADFS)
+
+> **Note:** Tested on ADFS 4.0, Windows Server 2019
```
Comment (matifali) [inline docs/admin/auth/adfs.md:3]:
```suggestion
> ![NOTE]
> Tested on ADFS 4.0, Windows Server 2019
```

File: docs/admin/auth/group-role-sync.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+```sh
+[debu]  coderd.userauth: got oidc claims  trace=0x1b09780  span=0x1b09820  request_id=833f136a-2e6b-4df5-8ecb-1316c71a425a  source=id_token  claim_fields="[aio aud email exp groups iat idp iss name nbf oid preferred_username rh sub tid uti ver]"  blank=[]
+
+[debu]  coderd.userauth: got oidc claims  trace=0x1b09780  span=0x1b09820  request_id=833f136a-2e6b-4df5-8ecb-1316c71a425a  source=userinfo  claim_fields="[email family_name given_name name picture sub]"  blank=[]
+
+[debu]  coderd.userauth: got oidc claims  trace=0x1b09780  span=0x1b09820  request_id=833f136a-2e6b-4df5-8ecb-1316c71a425a  source=merged  claim_fields="[aio aud email exp family_name given_name groups iat idp iss name nbf oid picture preferred_username rh sub tid uti ver]"  blank=[]
+```
+
+> ℹ️ In this example, Coder is successfully getting the `groups` OIDC claim from the token and merging the claims from userinfo endpoint. See below for troubleshooting instructions.
```
[the remark this answers is not in the document]
Comment (matifali) [reply docs/admin/auth/group-role-sync.md:39]: @BrunoQuaresma can these be rendered on coder.com/v2/docs

File: docs/admin/auth.md [hunk 6]
```diff
@@ -138,6 +136,74 @@ To upgrade Coder, run:
 helm upgrade <release-name> coder-v2/coder -n <namespace> -f values.yaml
 ```
 
+### Step 2 (JWT/PKI/Certificate Auth): Configure Coder with the OpenID Connect credentials
+
+<blockquote class="warning">
+  <p>
+  Only <b>Azure AD</b> has been tested with this method. Other OIDC providers may not work, as most providers add additional requirements ontop of the standard that must be implemented. If you are using another provider and run into issues, please leave an issue on our <a href="https://github.com/coder/coder/issues">Github</a>.
+  </p>
+</blockquote>
```
Comment (matifali) [inline docs/admin/auth.md:145]: I noticed that sometimes we use these HTML blockquotes and sometimes GitHub Style blockquote.

File: docs/admin/auth/passwordless.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# Passwordless Authentication
+
+You can create passwordless users in users for machine accounts. This can come in handy if you plan on [automating Coder](../automation.md) in CI/CD pipelines, for example.
+
+In the Users page `https://coder.example.com/users`, create a new user with `Login Type: none`:
+
+![Create new user](https://user-images.githubusercontent.com/22407953/262183871-9a9070fa-ca35-4816-9990-465b16b94fe4.png)
```
Comment (matifali) [inline docs/admin/auth/passwordless.md:7]: replace image path with realative

File: docs/admin/auth/oidc.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+If you'd like to change the OpenID Connect button text and/or icon, you can
+configure them like so:
+
+```console
+CODER_OIDC_SIGN_IN_TEXT="Sign with Azure AD"
+CODER_OIDC_ICON_URL="https://upload.wikimedia.org/wikipedia/commons/f/fa/Microsoft_Azure.svg"
+```
+
+![Custom OIDC text and icon](https://user-images.githubusercontent.com/22407953/261882846-1ce9c076-1247-4929-b082-72252b1f21c4.png)
```
Comment (matifali) [inline docs/admin/auth/oidc.md:110]: image path

**Review by matifali — APPROVED** [review]

File: docs/admin/auth.md [hunk 9]
```diff
@@ -1,7 +1,5 @@
 # Authentication
 
-[OIDC with Coder Sequence Diagram](https://raw.githubusercontent.com/coder/coder/138ee55abb3635cb2f3d12661f8caef2ca9d0961/docs/images/oidc-sequence-diagram.svg).
```
[the remark this answers is not in the document]
Comment (bpmct) [reply docs/admin/auth.md:3]: I link to it later in the doc under "How it works"

[the remark this answers is not in the document]
Comment (bpmct) [reply docs/admin/auth.md:3]: https://github.com/coder/coder/pull/9236/files#diff-8765e127bca65b250bca6b9256b4e8bffd123af134ec7c2f1e5a505a87219601R43

File: docs/admin/auth/keycloak.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Keycloak Authentication with Coder
+
+The access_type parameter has two possible values: "online" and "offline." By default, the value is set to "offline". This means that when a user authenticates using OIDC, the application requests offline access to the user's resources, including the ability to refresh access tokens without requiring the user to reauthenticate.
+
+To enable the `offline_access` scope, which allows for the refresh token functionality, you need to add it to the list of requested scopes during the authentication flow. Including the `offline_access` scope in the requested scopes ensures that the user is granted the necessary permissions to obtain refresh tokens.
```
Comment (Emyrk) [inline docs/admin/auth/keycloak.md:5]: Should we provide the example?

```console
CODER_OIDC_SCOPES=openid,profile,email,offline_access
```

File: docs/admin/auth/oidc.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+the claims from the ID token and the claims obtained from hitting the
+upstream provider's `userinfo` endpoint, and use the resulting data
+as a basis for creating a new user or looking up an existing user.
+
+To troubleshoot claims, set `CODER_VERBOSE=true` and follow the logs
+while signing in via OIDC as a new user. Coder will log the claim fields
+returned by the upstream identity provider in a message containing the
+string `got oidc claims`, as well as the user info returned.
+
+The following information is also avalible via a [sequence diagram](https://raw.githubusercontent.com/coder/coder/138ee55abb3635cb2f3d12661f8caef2ca9d0961/docs/images/oidc-sequence-diagram.svg) for visual learners.
```
Comment (Emyrk) [inline docs/admin/auth/oidc.md:55]: <3

Comment (Emyrk) [inline docs/admin/auth/oidc.md:55]: Change to `main` branch now imo.

```suggestion
The following information is also avalible via a [sequence diagram](https://raw.githubusercontent.com/coder/coder/main/docs/images/oidc-sequence-diagram.svg) for visual learners.
```

File: docs/admin/auth/oidc.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+### PKI Authentication (Optional)
+
+An alternative authentication method is to use signed JWT tokens rather than a shared `client_secret`. This requires 2 files.
+
+<blockquote class="warning">
+  <p>
+  Only <b>Azure AD</b> has been tested with this method. Other OIDC providers may not work, as most providers add additional requirements ontop of the standard that must be implemented. If you are using another provider and run into issues, please leave an issue on our <a href="https://github.com/coder/coder/issues">Github</a>.
+  </p>
+</blockquote>
```
Comment (Emyrk) [inline docs/admin/auth/oidc.md:120]: I know I probably did this, but move this up. The previous sentence and the content after this should be continous.

File: docs/admin/auth/passwordless.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+
+From there, you can create a long-lived token on behalf of the passwordless user using the [Create token API key](../../api/users.md#create-token-api-key):
+
+```sh
+# Replace API_KEY with a token from https://coder.example.com/cli-auth
```
Comment (Emyrk) [inline docs/admin/auth/passwordless.md:12]:
```suggestion
# Replace API_KEY with a token from https://coder.example.com/cli-auth
# The logged in user needs to have the requisite permissions to create an API
# key on behalf of another user.
```

File: docs/admin/auth/passwordless.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+curl -X POST http://coder-server:8080/api/v2/users/coder-bot/keys/tokens \
```
Comment (Emyrk) [inline docs/admin/auth/passwordless.md:13]:
```suggestion
curl -X POST http://coder-server:8080/api/v2/users/<machine_username>/keys/tokens \
```

**Review by Emyrk — APPROVED** [review]
I like this. I think we can always keep tweaking, but this is an improvement for sure.

File: docs/admin/auth.md (hunk 6 above)
Comment (matifali) [reply docs/admin/auth.md:145]: The other method is I am not sure if we support. I asked Bruno if he can add it.

File: docs/admin/auth/keycloak.md (hunk 10 above)
Comment (matifali) [reply docs/admin/auth/keycloak.md:5]: It would be better if we also include an example

File: docs/admin/auth/oidc.md (hunk 11 above)
Comment (matifali) [reply docs/admin/auth/oidc.md:55]: This is stored in docs/images. We plan to remove the absolute link and use relative link for images too.

**Comment by matifali** [comment]
A general comment is we should use `shell` for all code blocks that we expect users to copy and paste and `console` for where we are also showing the outputs of a command. 
@bpmct if you can change all `sh` to `shell` to be aligned with upcoming changes in #9245

File: docs/admin/auth/oidc.md (hunk 11 above)
Comment (Emyrk) [reply docs/admin/auth/oidc.md:55]: Sure

File: docs/admin/auth.md (hunk 1 above)
Comment (bpmct) [reply docs/admin/auth.md:99]: Im in favor of this?

File: docs/admin/auth/github.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+
+To restrict specific teams, set:
+
+```console
+CODER_OAUTH2_GITHUB_ALLOWED_TEAMS=frontend,backend
+```
+
+Once complete, restart your server. GitHub authentication should appear on your log-in screen 🎉
+
+![Log in with GitHub button](https://user-images.githubusercontent.com/22407953/261880741-87bfb1cb-7ea3-41b5-8eef-200552157cda.png)
```
[the remark this answers is not in the document]
Comment (bpmct) [reply docs/admin/auth/github.md:54]: Yeah I like to preview with the coder site too

**Pull request closed without merging**